pipeline_steps/checkmarx.groovy

```diff
@@ -68,7 +69,9 @@ def scan(String scan_type, String repo_name, String exclude_folders){
                 )
             } catch (Exception e){
                 print ("Caught exception while running checkmarx scan: "+e)
-                sleep(time: 30, unit: "SECONDS")
+                sleep(time: waitTime, unit: "SECONDS")
```

Comment:
If a whole load of jobs hit checkmarx, then start backing off, they'll all use the same backoff pattern and may hit the same contention issue when retrying. This could be mitigated by adding a small random value to the wait time. 

```suggestion
                sleep(time: waitTime + random.randrange(0, 60), unit: "SECONDS")
```

Probably have to import random for that to work.